File: src/main/java/com/lalkalol/mujahid/util/Identifiers.java
package com.lalkalol.mujahid.util;

import java.util.regex.Pattern;

public final class Identifiers {
    private static final Pattern URL = Pattern.compile("^https?://.*", Pattern.CASE_INSENSITIVE);

    private Identifiers() {
    }

    public static String fromQuery(String query) {
        String trimmed = query.trim();
        if (URL.matcher(trimmed).matches()) {
            return trimmed;
        }
        return "ytsearch:" + trimmed;
    }
}

File: src/main/kotlin/com/lalkalol/mujahid/util/Identifiers.kt
package com.lalkalol.mujahid.util

object Identifiers {
    private val URL = Regex("^https?://.*", RegexOption.IGNORE_CASE)

    fun fromQuery(query: String): String {
        val trimmed = query.trim()
        return if (URL.matches(trimmed)) trimmed else "ytsearch:$trimmed"
    }
}
